[Merge-Queue] Remove custom summaries when skipped

JonWBedard opened the pull request:
#### ddae38dc02f060f94cb0b3776863289d62a39421
<pre>
[Merge-Queue] Remove custom summaries when skipped
<a href="https://bugs.webkit.org/show_bug.cgi?id=238633">https://bugs.webkit.org/show_bug.cgi?id=238633</a>
&lt;rdar://problem/91125435 &gt;

Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateSquashed.getResultSummary):
(AddReviewerToCommitMessage.getResultSummary):
(AddReviewerToChangeLog.getResultSummary):
(ValidateCommitMessage.getResultSummary):
(Canonicalize.getResultSummary):
(PushPullRequestBranch.getResultSummary):
(UpdatePullRequest.getResultSummary):
* Tools/CISupport/ews-build/steps_unittest.py:

Canonical link: <a href="https://commits.webkit.org/249189@main">https://commits.webkit.org/249189@main</a>



Canonical link: <a href="https://commits.webkit.org/249189@main">https://commits.webkit.org/249189@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@292291">https://svn.webkit.org/repository/webkit/trunk@292291</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4638,7 +4638,7 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
     def getResultSummary(self):
         if self.results == SKIPPED:
-            return {'step': 'Patches are always squashed'}
+            return super(ValidateSquashed, self).getResultSummary()
```
This should be the default case. 
We should delete if condition for SKIPPED case, and change the current default return `Can only land squashed branches` to an explicit self.results value (e.g.: `if self.results == ERROR`).

There are many self.results values which are not covered by current default case (e.g.: WARNINGS, EXCEPTION), complete list here: https://github.com/buildbot/buildbot/blob/master/master/buildbot/process/results.py#L17

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4637,11 +4637,11 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
-            return {'step': 'Patches are always squashed'}
+        if self.results == FAILURE:
+            return {'step': 'Can only land squashed branches'}
         elif self.results == SUCCESS:
```
Nit: this elif can be if (since previous if statement has a return)

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4714,11 +4714,11 @@ def start(self, BufferLogObserverClass=logobserver.BufferLogObserver):
-            return {'step': 'No reviewer defined' if self.getProperty('github.number') else 'Patches have no commit message'}
+        if self.results == FAILURE:
+            return {'step': 'Failed to apply reviewers'}
         elif self.results == SUCCESS:
```
Ditto

aj062 commented on Tools/CISupport/ews-build/steps.py:
```diff
@@ -4762,11 +4762,11 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
-            return {'step': 'No reviewer defined' if self.getProperty('github.number') else 'Patches are edited upon application'}
+        if self.results == FAILURE:
+            return {'step': 'Failed to add reviewers to ChangeLogs'}
         elif self.results == SUCCESS:
```
Ditto

JonWBedard committed:
[Merge-Queue] Remove custom summaries when skipped
https://bugs.webkit.org/show_bug.cgi?id=238633
<rdar://problem/91125435>

Patch by Jonathan Bedard <jbedard@apple.com> on 2022-04-04
Reviewed by Aakash Jain.

* Tools/CISupport/ews-build/steps.py:
(ValidateSquashed.getResultSummary):
(AddReviewerToCommitMessage.getResultSummary):
(AddReviewerToChangeLog.getResultSummary):
(ValidateCommitMessage.getResultSummary):
(Canonicalize.getResultSummary):
(PushPullRequestBranch.getResultSummary):
(UpdatePullRequest.getResultSummary):
* Tools/CISupport/ews-build/steps_unittest.py:
